src/graphics/src/model.cpp

```diff
@@ -0,0 +1,112 @@
+#include <graphics/incl/model.h>
+#include <graphics/incl/mesh.h>
+#include <graphics/incl/shader_loader.h>
+
+Model::Model(ShaderLoader& loader, const char* path, 
+    const char* vertex_shader_path, const char* frag_shader_path) {
+    
+    LoadModel(path);
+    shader = loader.LoadShader(vertex_shader_path, frag_shader_path);
+}
+
+Model::~Model() {
+
+}
+
+void Model::Render(const glm::mat4& view, const glm::mat4& projection) {
+    for (auto mesh : meshes) {
+        //TODO: get actual model matrix lol
+        mesh.Render(glm::scale(glm::mat4(1.0f), glm::vec3(2.0f, 2.0f, 2.0f)) , view, projection, shader);
+    }
+}
+
+void Model::LoadModel(const std::string path) {
+    std::cout << "Loading model at path: " << path << std::endl;
+
+    // read file via ASSIMP
+    Assimp::Importer importer;
+    const aiScene* const scene = importer.ReadFile(path, aiProcess_Triangulate);
+
+    // check for errors
+    if (!scene || scene->mFlags == AI_SCENE_FLAGS_INCOMPLETE || !scene->mRootNode)
+    {
+        std::cout << "Error: Assimp could not load model -  " << importer.GetErrorString() << std::endl;
```

Comment:
Maybe we could add some perror here and future print statements to catch errors. I guess terror is implemented with built in knowledge meaning of errno lawl?